rpc/handlers.go

```diff
@@ -1537,7 +1560,7 @@ func (h *Handler) traceBlockTransactions(ctx context.Context, block *core.Block,
 		if errors.Is(err, utils.ErrResourceBusy) {
 			return nil, ErrInternal.CloneWithData(err.Error())
 		}
-		return nil, makeContractError(err)
+		return nil, ErrUnexpectedError.CloneWithData(err.Error())
```

Comment:
Is `ErrUnexpectedError` only for gateway errors?

Reply:
Trace is not expected to return ContractError anymore.
When tracing an existing block, we expect vm.Execute to never return an error. Since we know sequencer executed those transactions just fine. Hence, the `ErrUnexpectedError` if vm.Execute returns an unexpected error.

Reply:
Ah, that's very good reasoning. Mind adding that in a comment? It doesn't immediately occur to me when reading the code.